picard/metadata.py

```diff
@@ -290,49 +320,62 @@ def clear_deleted(self):
         self.deleted_tags = set()
 
     def getall(self, name):
-        return super().get(name, [])
+        return self._store.get(name, [])
+
+    def getraw(self, name):
+        return self._store[name]
 
     def get(self, name, default=None):
-        values = super().get(name, None)
+        values = self._store.get(name, None)
         if values:
             return self.multi_valued_joiner.join(values)
         else:
             return default
 
+    def __contains__(self, name):
+        return self._store.__contains__(name)
+
     def __getitem__(self, name):
         return self.get(name, '')
 
     def set(self, name, values):
-        super().__setitem__(name, values)
-        if name in self.deleted_tags:
-            self.deleted_tags.remove(name)
+        self._store[name] = values
+        self.deleted_tags.discard(name)
 
     def __setitem__(self, name, values):
         if not isinstance(values, list):
             values = [values]
         values = [str(value) for value in values if value]
```

Comment:
If we already touch this we should probably take care of the inconsistency that `__setitem__` only stores string values and `add` stores the values as is. E.g. 

```python
m["sometag"] = 1  # Stored as string "1"
m.add("sometag", 1)  # Stored as integer 1
```

Actually having the tags treated as strings consistently would allow us to eliminate quite a few type checks and string conversions currently done when setting and reading tags. There might come probably some edge cases where we would like to store non-string values, but those should be rare. As soon as tags get handled by tagger script they get converted into strings anyway. If there will be some need we could still it some `setraw` method or similar.

Reply:
Yes, for now, i tried to preserve the current behavior, but, clearly, the goal is to make things much more documented and consistent.
And some stuff can be removed (like `Metadata.delete()` which is redundant with `Metatada.__delitem__()` which, as bonus, provide `del m['tag']` syntax, which is more natural imho)

I don't want to start any modification yet, the idea is to first merge this, than start to clean up code actually using `Metadata` (i was forced to change some tests because of `dict.get()` hacky stuff, but this PR has no other changes unrelated to this refactoring).